qiskit/compiler/assembler.py

```diff
@@ -243,24 +290,135 @@ def _parse_common_args(backend, qobj_id, qobj_header, shots,
                            memory=memory,
                            max_credits=max_credits,
                            seed_simulator=seed_simulator,
                            init_qubits=init_qubits,
                            rep_delay=rep_delay,
+                           qubit_lo_freq=qubit_lo_freq,
+                           meas_lo_freq=meas_lo_freq,
+                           qubit_lo_range=qubit_lo_range,
+                           meas_lo_range=meas_lo_range,
+                           schedule_los=schedule_los,
+                           n_qubits=n_qubits,
                            **run_config)
 
     return qobj_id, qobj_header, run_config_dict
 
 
-def _parse_pulse_args(backend, qubit_lo_freq, meas_lo_freq, qubit_lo_range,
-                      meas_lo_range, schedule_los, meas_level,
-                      meas_return, meas_map,
-                      memory_slot_size,
-                      rep_time, parametric_pulses,
-                      **run_config):
+def _check_lo_freqs(
+    qubit_lo_freq: Union[None, List[float]],
+    meas_lo_freq: Union[None, List[float]],
+    qubit_lo_range: Union[None, List[List[float]]],
+    meas_lo_range: Union[None, List[List[float]]],
+    n_qubits: Union[None, int],
+):
+    """Check lo frequency arrays are the correct size and in range.
+
+    Args:
+        qubit_lo_freq: List of qubit lo frequencies.
+        meas_lo_freq: List of meas lo frequencies.
+        qubit_lo_range: List of allowed qubit lo frequencies. Each element is 2d list of form
+            ``[min_qubit_lo_freq, max_qubit_lo_freq]``.
+        meas_lo_range: List of allowed meas lo frequencies. Each element is 2d list of form
+            ``[min_meas_lo_freq, max_meas_lo_freq]``.
+
+    """
+    # only check if is a valid number of qubits in system
+    if not n_qubits:
+        return
+
+    _check_lo_range(qubit_lo_freq, qubit_lo_range, "qubit", n_qubits)
+    _check_lo_range(meas_lo_freq, meas_lo_range, "meas", n_qubits)
+
+
+def _check_lo_length(lo_list: Union[List[float], None], list_type: str, n_qubits: int):
+    """Check that lo lists have length equal to the number of qubits in the system.
+
+    Args:
+        lo_list: List involving lo frequencies--either the freqs themselves or the range of allowed
+            freqs.
+        list_type: Type of lo list--"qubit_lo_freq", "meas_lo_freq", "qubit_lo_range",
+            "meas_lo_range".
+        n_qubits: Number of qubits in the system.
+    """
+    if lo_list and len(lo_list) != n_qubits:
+        raise QiskitError(
+            "Length of {} is {}. It must be equal to {}, the number of qubits on "
+            "this backend.".format(list_type, len(lo_list), n_qubits)
+        )
+
+
+def _check_lo_range(
+    lo_freq: Union[List[float], None],
+    lo_range: Union[List[float], None],
+    lo_type: str,
+    n_qubits: int,
+):
+    """Check that lo freqs are within the perscribed lo range. Also checks length of both lo lists.
```

Comment:
```suggestion
    """Check that LO frequencies are within the prescribed LO range. Also checks the length of both LO lists.
```

Reply:
Addressed in https://github.com/Qiskit/qiskit-terra/pull/6167/commits/705ca3c97ae84c6f950f430f1eea576d4eb93e1c.